File: 🟡 Medium/1405 - Longest Happy String/_1405LongestHappyString.cpp
class Solution {
public:
    string longestDiverseString(int a, int b, int c) {
        int currA = 0, currB = 0, currC = 0;
        int maxLen = a + b + c;
        int i = 0;
        string sb = "";
        while (i < (maxLen)) {
            if (currA != 2 && a >= b && a >= c ||
                a > 0 && (currB == 2 || currC == 2)) {
                sb += 'a';
                currA++;
                currB = 0;
                currC = 0;
                a--;
            }

            else if (currB != 2 && b >= a && b >= c ||
                     b > 0 && (currA == 2 || currC == 2)) {
                sb += 'b';
                currB++;
                currA = 0;
                currC = 0;
                b--;
            }

            else if (currC != 2 && c >= a && c >= b ||
                     c > 0 && (currA == 2 || currB == 2)) {
                sb += 'c';
                currC++;
                currA = 0;
                currB = 0;
                c--;
            }
            i++;
        }
        return sb;
    }
};

File: 🟡 Medium/1405 - Longest Happy String/_1405LongestHappyString.py
class Solution:
    def longestDiverseString(self, a: int, b: int, c: int) -> str:
        currA = 0
        currB = 0
        currC = 0
        maxLen = a + b + c
        i = 0
        sb = ""
        for i in range(0, maxLen):
            if currA != 2 and a >= b and a >= c or a > 0 and (currB == 2 or currC == 2):
                sb += "a"
                currA += 1
                currB = 0
                currC = 0
                a -= 1

            elif (
                currB != 2 and b >= a and b >= c or b > 0 and (currA == 2 or currC == 2)
            ):
                sb += "b"
                currB += 1
                currA = 0
                currC = 0
                b -= 1

            elif (
                currC != 2 and c >= a and c >= b or c > 0 and (currA == 2 or currB == 2)
            ):
                sb += "c"
                currC += 1
                currA = 0
                currB = 0
                c -= 1

        return sb
